source/dub/dependency.d

```diff
@@ -21,12 +21,28 @@ import std.exception;
 import std.string;
 import std.sumtype;
 
+/** Encapsulates the name of a package.
+*/
+struct PackageName {
+	string value;
+	@disable this(PackageName name) @safe pure nothrow @nogc;
+	this(string value) @safe pure nothrow @nogc {
+		this.value = value;
+	}
+	static typeof(this) fromString(string value) @safe pure nothrow @nogc {
+		return typeof(return)(value);
+	}
+	string toString() @property @trusted const scope {
+		return value.color(Mode.bold);
+	}
```

Comment:
I think one should not be able to construct a `PackageName` with an invalid string. You might also want to implement `opAssign` and move the splitting logic (and whether or not the name is that of a subpackage) here.
See the linked commit as an example.

Reply:
I don't understand the part about the opAssign. Can you elaborate on why we need `PackageName.opAssign`?

Reply:
Yes. Otherwise we bypass the check as we have `alias this`.